server/job_test.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"path/filepath"
+	"testing"
+
+	"github.com/mattermost/mattermost/server/v8/platform/shared/filestore"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDeleteDump(t *testing.T) {
+	fs, err := filestore.NewFileBackend(filestore.FileBackendSettings{
+		DriverName: "local",
+		Directory:  t.TempDir(),
+	})
+	require.NoError(t, err)
+
+	plugin := &Plugin{
+		fileBackend: fs,
+	}
+
+	t.Run("Successfully delete dump directory", func(t *testing.T) {
+		job := &DumpJob{
+			DumpLocation: filepath.Join(pluginDataDir, PluginName, "some-dump"),
+		}
+
+		success, err := job.DeleteDump(plugin)
+		require.NoError(t, err)
+		require.True(t, success)
+	})
+
+	t.Run("Dump location not under plugin-data directory", func(t *testing.T) {
+		job := &DumpJob{
+			DumpLocation: "/some-other-dir/some-dump",
+		}
+
+		success, err := job.DeleteDump(plugin)
+		require.NoError(t, err)
+		require.False(t, success)
+	})
+
+	t.Run("Empty dump location", func(t *testing.T) {
+		job := &DumpJob{
+			DumpLocation: "",
+		}
+
+		success, err := job.DeleteDump(plugin)
+		require.NoError(t, err)
+		require.False(t, success)
+	})
```

Comment:
Would it be a good idea to check if the `fs` was asked to delete anything?

Reply:
Added an assertion for that, created a temp file and then checked if it was deleted or not.